src/components/dataPanel/EditPanelButton.tsx

```diff
@@ -1,12 +1,19 @@
 import { Button, Typography } from "@material-ui/core";
+import { IEntityLink } from "entities/IEntityLink";
 import React, { FunctionComponent } from "react";
 
 export const EditPanelButton: FunctionComponent<{
   onClick: () => void;
-}> = ({ onClick, children }): JSX.Element => (
-  <Typography align="right" style={{ paddingRight: "5em" }}>
-    <Button variant="outlined" onClick={onClick}>
-      {children}
-    </Button>
-  </Typography>
-);
+  links?: IEntityLink[];
+}> = ({ onClick, links, children }): JSX.Element => {
+  const canEdit = links == null || links.some((link) => link.verb === "PATCH");
+  return canEdit ? (
+    <Typography align="right" style={{ paddingRight: "5em" }}>
+      <Button variant="outlined" onClick={onClick}>
+        {children}
+      </Button>
+    </Typography>
+  ) : (
+    <></>
```

Comment:
I'm wondering if this is the best place to have the `allow user to edit` logic.  The `<></>` at line 17 makes me curious...

Why did you choose to put the logic here and not in the parent panel, for example?  I can think of pros/cons either way!  Keen to understand your thinking.

Reply:
it's in a single place here, if we put it above the button on the panel you'd have to repeat the logic for each panel types, BeaconPanel, BeaconUsePanel etc.

I think 🤔

Reply:
Good point, works for me!